pycompwa/data/convert.py

```diff
@@ -0,0 +1,118 @@
+"""Conversion tools between from ComPWA objects and :class:`pandas.DataFrame`.
+
+This module contains functions that help convertion from ComPWA objects like
+:class:`~.EventCollection` and :class:`~.DataSet` to a
+:class:`~pandas.DataFrame` and back.
+"""
+
+
+__all__ = [
+    'data_set_to_pandas',
+    'events_to_pandas',
+    'pandas_to_events',
+]
+
+
+import pandas as pd
+
+import pycompwa.ui as pwa
+
+from . import _labels
+from . import create
+from . import exception
+from . import kinematics
+from . import naming
+
+
+def events_to_pandas(
+        events: pwa.EventCollection, model: str = None,
+        compute_kinematics: bool = False) -> pd.DataFrame:
+    """Convert an `~.EventCollection` to a  `~pandas.DataFrame`.
+
+    Create a PWA formatted `~pandas.DataFrame` from an
+    `~.EventCollection`.
+
+    Parameters:
+        events: The `~.EventCollection` that you want to convert.
+        model: Name of the XML file containing the kinematic info or a
+            `.Kinematics` instance.
+        compute_kinematics: Directly compute the kinematic variables for each
+            momentum tuple. If ``True``, you have to specify ``model``!
+    Raises:
+        ConfigurationConflict: Number of final state particles in XML does not
+            agree with number of final state particles in `~.PwaAccessor`.
+        MissingParameter: If ``model`` was not specified.
+    """
+    pids = events.pids
+    if model:
+        id_to_name = kinematics.id_to_name_map(model)
+        if len(id_to_name) != len(pids):
+            raise exception.ConfigurationConflict(
+                f'XML file {model} has {len(id_to_name)} final state'
+                f'particles, the event collection has {len(pids)}')
+        particle_list = pwa.read_particles(model)
+        pids = [particle_list.name_to_pid(name)
+                for name in id_to_name.values()]
+    particles = naming.make_values_unique(pids)
+    columns = create.multicolumn(particles)
+    frame = pd.DataFrame(events.to_table(), columns=columns)
+    if model:
+        particle_list = pwa.read_particles(model)
+        naming.pid_to_name(frame, particle_list)
+    if events.has_weights():
+        frame[_labels.WEIGHT] = events.weights()
+    if compute_kinematics:
+        if model is None:
+            raise exception.MissingParameter(
+                "XML model file required in order to compute kinematic"
+                "variables")
+        data_set = kinematics.compute(events, model)
+        for var in data_set.data.keys():
+            frame[var] = data_set.data[var]
+    return frame
+
+
+def pandas_to_events(
```

Comment:
I would not implement such a function because I don't see a use case for it. Its just additional code that needs to be written and tested.

Reply:
Please see [`data_module.ipynb`](https://github.com/redeboer/pycompwa/blob/9b9bece72007eabc9af1f0d485096c1694888a1e/examples/tools/data_module.ipynb)

Reply:
> I would not implement such a function because I don't see a use case for it.

@Leongrim ?

Reply:
Apart from this I do not have any more issues with the PR

Reply:
Maybe @Leongrim or Sebastian can comment on this.

Reply:
I think it is a good way to convert python/pandas to ComPWA.
For instance I had often the problem that I wanted to calculate angle distributions for particles (which I had already in pandas anyway) and the way to first create a root file and then load it is cumbersome.

Reply:
https://github.com/ComPWA/tensorwaves/issues/18#issuecomment-615268099

Reply:
ok